### ISSUE TITLE
terraform get hanging since 0.6.7 upgrade

### ISSUE THREAD

**Issue by nickithewatt** [hashicorp/terraform#4082]
Hi, 

since upgrading to 0.6.7, my ```terraform get``` seems to be hanging and never returning. I am running on Mac OS X, and have deleted my .terraform directory. I do have two different modules pointing to the same git module source locations (a tagged sub directory), however this same setup works fine in 0.6.6.
 
````
module "mod1" {
    source = "git::ssh://git@gitrepo.org/team/reponame//src/terraform/modules/openstack?ref=0.0.1"
    ...
}
module "mod2" {
    source = "git::ssh://git@gitrepo.org/team/reponame//src/terraform/modules/openstack?ref=0.0.1"
    ...
}
````

**Comment by xsmaster** [comment]
What version of OSX are you on? I have seen similar issues with 10.11.1 El Capitan.

**Comment by nickithewatt** [comment]
I am still on OS X Yosemite (10.10.5), unfortunately turning debug does not show anything new, it still just hangs at the ```get``` command and never returns, I have to kill -9 it to get it back

**Comment by xsmaster** [comment]
The below will help to kill all the instances of terraform. I did the mistake of hitting "Control + c " in the shell to kill it. It left a ton of things open.
```
kill $(ps aux | grep 'terraform' | awk '{print $2}')
```

So I setup the export and run it again with no help. One thing I did notice between 0.6.6 & 0.6.7 is that in 0.6.6 makes an outbound API call to hashi corp and 0.6.7 does not. 

https://checkpoint-api.hashicorp.com/v1/check/terraform?arch=amd64&os=darwin&signature=572dde7a-b371-b130-2423-7ca8aa28453a&version=0.6.6

GET /v1/check/terraform?arch=amd64&os=darwin&signature=572dde7a-b371-b130-2423-7ca8aa28453a&version=0.6.6 HTTP/1.1
Host: checkpoint-api.hashicorp.com
User-Agent: HashiCorp/go-checkpoint
Accept: application/json
Accept-Encoding: gzip

POST
HTTP/1.1 200 OK
Server: Cowboy
Connection: keep-alive
Access-Control-Allow-Headers: Accept, Cache-Control, X-Requested-With, Content-Type
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 86400
Content-Type: application/json
Date: Fri, 27 Nov 2015 04:47:33 GMT
Content-Length: 302
Via: 1.1 vegur

{"product":"terraform","current_version":"0.6.7","current_release":1448312818,"current_download_url":"https://terraform.io/downloads.html","current_changelog_url":"https://github.com/hashicorp/terraform/blob/v0.6.7/CHANGELOG.md","outdated":true,"project_website":"https://www.terraform.io","alerts":[]}

While writing this post it seems the API call is working now for me. before it was not checking the version.

https://checkpoint-api.hashicorp.com/v1/check/terraform?arch=amd64&os=darwin&signature=5778c68c-0c2a-78ee-b111-8896874bcc73&version=0.6.7

**Comment by xsmaster** [comment]
@nickithewatt I think i found the issue. if ANY of your .tf files have a # on the last line it will break terraform. 

Example: 

```
# Specify the provider and access details
provider "aws" {
  region = "${var.aws_region}"
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
}
#
```

From the command line type in:

```
terraform plan
```

Now it will just hang and do nothing. Once I removed all the ending # 0.6.7 works fine. this was not an issue in 0.6.6

**Comment by nickithewatt** [comment]
@xsmaster so that helped with some progress thanks! Having removed the trailing comment line from one of my files, I can at least Ctrl-C (SIGINT) stop the ```terraform get``` without resorting to having to kill -9 it, however it still hangs for me.

**Comment by jen20** [comment]
Hi @nickithewatt! Thanks for reporting this. There is another report of `terraform get` behaviour not working in all cases as well; clearly this is something we need to address. @xsmaster nice sleuthing to find the issue with trailing `#` - I think this is a bug in the new HCL parser, and should be relatively straightforward to fix.

**Comment by nickithewatt** [comment]
Great thanks @jen20 and @xsmaster!

**Comment by xsmaster** [comment]
Glad to help. I love terraform

**Comment by antonbabenko** [comment]
Just saying. I had same issue on 0.6.7, and have just verified that the problem is fixed on master `Terraform v0.6.8-dev (e80ad0266edc7bb0e480f3939ff4801aed7448d1)`.

**Comment by phinze** [comment]
Should be fixed w/ the v0.6.8 release! :tada:

**Comment by xsmaster** [comment]
Win!!!
